aten/src/ATen/native/quantized/cpu/fbgemm_utils.h

```diff
@@ -32,14 +33,96 @@ struct CAFFE2_API PackedLinearWeightFp16 {
 };
 
 template <int kSpatialDim = 2>
-struct CAFFE2_API PackedConvWeight {
+struct CAFFE2_API PackedConvWeight : public ConvPackedParamsBase<kSpatialDim> {
+  PackedConvWeight(
+      std::unique_ptr<fbgemm::PackWeightsForConv<kSpatialDim>> w,
+      c10::optional<at::Tensor> bias,
+      torch::List<int64_t> stride,
+      torch::List<int64_t> padding,
+      torch::List<int64_t> dilation,
+      int64_t groups,
+      std::vector<int32_t> col_offsets,
+      std::vector<int64_t> kernel,
+      std::vector<float> w_scale,
+      std::vector<int32_t> w_zp,
+      c10::QScheme q_scheme)
+    : w(std::move(w)),
+    bias(bias),
+    stride_(stride),
+    padding_(padding),
+    dilation_(dilation),
+    groups_(groups),
+    col_offsets(col_offsets),
+    kernel(kernel),
+    w_scale(w_scale),
+    w_zp(w_zp),
+    q_scheme(q_scheme) {}
+
   std::unique_ptr<fbgemm::PackWeightsForConv<kSpatialDim>> w;
   c10::optional<at::Tensor> bias;
+  torch::List<int64_t> stride_;
+  torch::List<int64_t> padding_;
+  torch::List<int64_t> dilation_;
+  int64_t groups_;
   std::vector<int32_t> col_offsets;
   std::vector<int64_t> kernel;
   std::vector<float> w_scale;
   std::vector<int32_t> w_zp;
   c10::QScheme q_scheme;
+
+  at::Tensor apply(
+      at::Tensor input,
+      double output_scale,
+      int64_t output_zero_point) override {
+    return apply_impl<false>(input, output_scale, output_zero_point);
```

Comment:
Question: will this `apply_impl` callsite cause instantiation in compilation units other than `qconv.cpp`? I see you've instantiated `apply_imp` in `qconv.cpp`, but should the definitions of `apply` and `apply_relu` be in that file as well?

Reply:
right, I think these can be removed

Reply:
looks like we can remove all explicit template instantiations in qconv.cpp

Reply:
Looks like we still need them

Reply:
moved these to qconv.cpp